src/Security.tsx

```diff
@@ -60,6 +61,10 @@ const Security: React.FC<{
       restoreOriginalUri(oktaAuth as OktaAuth, originalUri);
     }) as ((oktaAuth: OktaAuth, originalUri?: string) => Promise<void>);
 
+    return () => {
+      // Restore original callback
+      oktaAuth.options.restoreOriginalUri = originalCallback;
```

Comment:
I don't think this is necessary, the `Security` component should never unmount

Reply:
This cleanup function fixes the issue with React 18 StrictMode.
If you run `test/apps/v6-app` sample app without it, you can notice the warning ```Two custom restoreOriginalUri callbacks are detected. The one from the OktaAuth configuration will be overridden by the provided restoreOriginalUri prop from the Security component.```
It happens because effects run twice

Reply:
Oh interesting, that's probably because StrictMode was changed in React18 to do a double mount

I wonder if we should use this approach instead: https://react.dev/learn/you-might-not-need-an-effect#initializing-the-application

Trevor used it as well for something similar: https://github.com/okta/okta-react/pull/266

Reply:
Changed to global var approach.
This required a workaround for tests: https://github.com/okta/okta-react/pull/275/commits/0132a0e0f565da359830b1f03cf5db254ee51854